WordPress/Helpers/RulesetPropertyHelper.php

```diff
@@ -43,15 +43,24 @@ final class RulesetPropertyHelper {
 	 * @since 2.0.0  No longer supports custom array properties which were incorrectly
 	 *               passed as a string.
 	 * @since 3.0.0  Moved from the Sniff class to this class.
+	 * @since 3.1.0  Added a new parameter to lowercase array keys and values.
 	 *
-	 * @param array $custom Custom list as provided via a ruleset.
-	 * @param array $base   Optional. Base list. Defaults to an empty array.
-	 *                      Expects `value => true` format when `$flip` is true.
-	 * @param bool  $flip   Optional. Whether or not to flip the custom list.
-	 *                      Defaults to true.
+	 * @param array $custom             Custom list as provided via a ruleset.
+	 * @param array $base               Optional. Base list. Defaults to an empty array.
+	 *                                  Expects `value => true` format when `$flip` is true.
+	 * @param bool  $flip               Optional. Whether or not to flip the custom list.
+	 * @param bool  $lowercaseKeyValues Optional. Whether to lowercase keys and values in the resulting array.
```

Comment:
```suggestion
	 * @param bool  $lowercasekeys Optional. Whether to lowercase keys in the resulting array.
```

Lowercasing the keys for use with an `isset()` lookup is one thing, lowercasing the values is something completely different and should **not** be done.

Using `strtolower()` on the values will change the type for any non-string scalars to string and will throw fatal errors for non-scalars.

Reply:
I think I added that because in the `NoSilencedErrors` sniff the `customAllowedFunctionsList` is having its values lowercased (https://github.com/WordPress/WordPress-Coding-Standards/blob/develop/WordPress/Sniffs/PHP/NoSilencedErrorsSniff.php#L187-L188). 
So I lowercased the values in the helper, seeing how in the majority of cases the non-list arrays have booleans for the values. 

I'll try to see what the values are in every case where this helper is used and see if it's safe to leave it, otherwise, I'll remove that part and revert the change in the `NoSilencedErrors` sniff.

Reply:
Values coming from properties are (in most cases) "flipped" to be the keys via the helper function, so the resulting array can be used with `isset()`.

I haven't checked the code, but might well be that what is being done in the NoSilencedErrors sniff is basically the same as you're doing in this PR, it just was already done for that sniff ?

Reply:
From what I've seen it was just done to this one sniff so it might be an isolated case. I'll need to take a bit closer look.